app/models/journal_cutoff_date.rb

```diff
@@ -0,0 +1,73 @@
+class JournalCutoffDate < ActiveRecord::Base
+  include ActiveModel::ForbiddenAttributesProtection
+
+  validates :cutoff_date, presence: true
+  validate :unique_month_for_cutoff_date, if: :cutoff_date?
+
+  def self.recent_and_upcoming
+    where("cutoff_date > ?", 1.month.ago.beginning_of_month).order(:cutoff_date)
+  end
+
+  def self.past
+    where("cutoff_date < ?", Time.current).order(:cutoff_date)
+  end
+
+  def self.first_valid_date
+    past.last.try(:cutoff_date).try(:beginning_of_month)
+  end
+
+  def last_valid_date
+    cutoff_date.prev_month.end_of_month
+  end
+
+  def past?
+    cutoff_date < Time.current
+  end
+
+  # TODO consider turning this into a reusable module
+  def cutoff_date_time=(hash)
+    formatted = "#{cutoff_date.to_date} #{hash[:hour]}:#{hash[:minute]}#{hash[:ampm]}"
+    self.cutoff_date = Time.zone.parse(formatted)
+  end
```

Comment:
I've always felt kind of weird of where input parsing (especially into date/times) should happen. On one hand, I want the model to be able to support `assign/update_attributes` with the `params` so that the controller stays normal. On the other hand, yeah, it's a little weird to do it in the model.

I've done things in the past where [I include a module and then call a method from that module](https://github.com/tablexi/rogerebert/blob/develop/lib/formattable_date_fields.rb). something like `datetime_attribute :cutoff_time, :some_other_time_field` that would set up extra methods like this, but I've never been fully satisfied with the results, but I've yet to find a better way. (It especially gets weird if the app is in UTC--NUcore is in Central Time).